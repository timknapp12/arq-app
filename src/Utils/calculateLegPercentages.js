// this function is to calculate the max QOV per leg based on what the
// current user's QOV is per each leg

export const calculateLegPercentages = (user, requirements) => {
  let leg1Max = requirements.legMaxOV;
  let leg2Max = requirements.legMaxOV;
  let leg3Max;
  // lower levels with 60% max
  if (requirements.legMaxPerc === 60) {
    // start with 100% of possible QOV per required rank
    let remainingPerc = 100;
    // get the user's QOV percentage of FIRST leg of the requirement
    let perc1 = ((user.leg1OV / requirements.requiredQOV) * 100).toFixed(0);
    if (perc1 >= 60) {
      perc1 = 60;
      remainingPerc -= perc1;
    } else {
      remainingPerc -= perc1;
    }
    // calculate the max possible QOV of the SECOND leg based on percentage of first leg
    leg2Max =
      remainingPerc >= 60
        ? requirements.legMaxOV
        : requirements.requiredQOV * (remainingPerc / 100);
    // get the user's QOV percentage of SECOND leg of the requirement
    let perc2 = ((user.leg2OV / requirements.requiredQOV) * 100).toFixed(0);
    if (perc2 >= 40) {
      perc2 = 40;
      remainingPerc -= perc2;
    } else {
      remainingPerc -= perc2;
    }
    // calculate the max possible QOV of the THIRD leg based on percentage of first leg
    leg3Max = requirements.requiredQOV * (remainingPerc / 100);
    return {
      leg1Max,
      leg2Max,
      leg3Max,
    };
  } else {
    // higher levels with 50% or 40% max - basically the same as above but only the THIRD leg possible QOV needs to be calculated
    let remainingPerc = 100;
    let perc1 = ((user.leg1OV / requirements.requiredQOV) * 100).toFixed(0);
    if (perc1 >= requirements.legMaxPerc) {
      perc1 = requirements.legMaxPerc;
      remainingPerc -= perc1;
    } else {
      remainingPerc -= perc1;
    }
    let perc2 = ((user.leg2OV / requirements.requiredQOV) * 100).toFixed(0);
    if (perc2 >= requirements.legMaxPerc) {
      perc2 = requirements.legMaxPerc;
      remainingPerc -= perc2;
    } else {
      remainingPerc -= perc2;
    }
    leg3Max = requirements.requiredQOV * (remainingPerc / 100);
    return {
      leg1Max,
      leg2Max,
      leg3Max,
    };
  }
};
