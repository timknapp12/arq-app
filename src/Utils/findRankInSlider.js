export const findRankInSlider = (list, value) => {
  const result = list.find((_, index) => index === value);
  return result.name;
};
