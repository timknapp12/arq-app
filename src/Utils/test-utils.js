import React from 'react';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react-native';
import AppContext from '../Contexts/AppContext';
import { ThemeProvider } from 'styled-components/native';
import { darkTheme } from '../Styles/themes';

const setIsSignedIn = jest.fn();

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={darkTheme}>
      <AppContext.Provider value={{ theme: darkTheme, setIsSignedIn }}>
        {children}
      </AppContext.Provider>
    </ThemeProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

AllTheProviders.propTypes = {
  children: PropTypes.node,
};
// re-export everything
export * from '@testing-library/react-native';

// override render method
export { customRender as render };
