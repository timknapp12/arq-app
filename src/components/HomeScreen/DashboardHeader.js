import React from 'react';
import PropTypes from 'prop-types';
import Header from '../Header';
import {
  H2Normal,
  Flexbox,
  SmallQIcon,
  BellIcon,
  AccountIcon,
} from '../Common';

const DashboardHeader = ({ badgeValue }) => {
  return (
    <Header>
      <Flexbox width="60px" align="flex-start">
        <SmallQIcon />
      </Flexbox>
      <H2Normal>Business</H2Normal>
      <Flexbox width="60px" direction="row">
        <BellIcon badgeValue={badgeValue} />
        <AccountIcon />
      </Flexbox>
    </Header>
  );
};

DashboardHeader.propTypes = {
  badgeValue: PropTypes.number,
};

export default DashboardHeader;
