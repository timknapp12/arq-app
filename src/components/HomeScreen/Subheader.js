import React from 'react';
import styled from 'styled-components/native';

//TODO = storybook

const Container = styled.View`
  width: 100%;
  height: 48px;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  padding: 0 24px;
  background-color: ${(props) => props.theme.inactiveBackground};
`;

const PlaceholderTxt = styled.Text`
  color: white;
`;

const Subheader = () => {
  return (
    <Container>
      <PlaceholderTxt>Overview</PlaceholderTxt>
      <PlaceholderTxt>Rank</PlaceholderTxt>
      <PlaceholderTxt>OV Detail</PlaceholderTxt>
    </Container>
  );
};

export default Subheader;
