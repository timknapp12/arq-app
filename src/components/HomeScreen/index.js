import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { ScreenContainer, H2, PrimaryButton } from '../Common';
import AppContext from '../../Contexts/AppContext';

const HomeScreen = ({ navigation }) => {
  const { setIsSignedIn } = useContext(AppContext);
  return (
    <ScreenContainer>
      <H2 testID="home-screen-title">Home Screen</H2>
      <PrimaryButton
        testID="go-to-profile-button"
        onPress={() => navigation.navigate('ProfileScreen')}>
        See Profile
      </PrimaryButton>
      <PrimaryButton
        style={{ marginTop: 12 }}
        onPress={() => setIsSignedIn(false)}>
        Sign out
      </PrimaryButton>
    </ScreenContainer>
  );
};

HomeScreen.propTypes = {
  navigation: PropTypes.object.isRequired,
};

export default HomeScreen;
