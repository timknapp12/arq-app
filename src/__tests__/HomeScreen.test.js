import React from 'react';
import { render } from '../Utils/test-utils';

import HomeScreen from '../Components/HomeScreen';

describe('<HomeScreen />', () => {
  it('Home Screen has a title called Home Screen', async () => {
    const { getByTestId, toJSON } = render(<HomeScreen navigation={{}} />);

    const screenTitle = 'Home Screen';

    expect(getByTestId('home-screen-title').props.children).toBe(screenTitle);
    expect(toJSON()).toMatchSnapshot();
  });
});
