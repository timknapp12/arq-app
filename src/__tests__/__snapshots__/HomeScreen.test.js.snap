// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> Home Screen has a title called Home Screen 1`] = `
Array [
  "/",
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": 0,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "height": "100%",
          "justifyContent": "center",
          "paddingTop": 60,
          "width": "100%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 24,
          },
        ]
      }
      testID="home-screen-title"
    >
      Home Screen
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "opacity": 1,
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
          "width": "100%",
        }
      }
      testID="go-to-profile-button"
    >
      <Text
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 18,
              "textAlign": "center",
            },
          ]
        }
      >
        See Profile
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "marginTop": 12,
          "opacity": 1,
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 18,
              "textAlign": "center",
            },
          ]
        }
      >
        Sign out
      </Text>
    </View>
  </View>,
]
`;
